cpl_replacement_cache.go

```diff
@@ -0,0 +1,98 @@
+package kbucket
+
+import (
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	"github.com/wangjia184/sortedset"
+)
+
+// TODO Should ideally use a Circular queue for this
+// maintains a bounded, de-duplicated and FIFO peer candidate queue for each Cpl
+type cplReplacementCache struct {
+	localPeer    ID
+	maxQueueSize int
+
+	sync.Mutex
+	candidates map[uint]*sortedset.SortedSet // candidates for a Cpl
```

Comment:
The sorted set code looks pretty complicated for what we need. I'd honestly just keep a single `map[uint][]peer.ID`. Given that we only have 20 slots per CPL, looking up the right bucket and checking every peer will be very cheap (compared to the SHA256 we'll do to actually _get_ the common prefix length).

Reply:
I wanted to use this because it makes it easy to change this code to store candidates based on some of _scoring_ in the _future_. However, you are right. We only have FIFO for now & can cross that bridge when we get to it.

Made the change.